### PR TITLE
Make sections in Resolution Results dialog resize when collapsed #1611

### LINKED ISSUE #1611: Collapsible sections in Resolution Results dialog don't work
The _Resolution Results_ ( has two collapsible sections named _Optional Resources_ and _Reasons_. However, collapsing a section does not cause the surrounding controls to take advantage from the free space. See the screenshots:

Expanded:
image

Collapsed:
image

BTW, I think it is discouraged to use Forms controls and look & feel in dialogs. To be consistent with other Eclipse plug-ins, Twisties like the ones from the _Java Compiler_ > _Errors/Warnings_ project pererence dialog should be used.

### PR DESCRIPTION
**Pull request by wodencafe**
This makes the uncollapsed sections in Resolution Results Dialog resize when one of them is collapsed.

This fixes #1611.

Signed-off-by: Chris Boyd <wodencafe@gmail.com>

### CODE REVIEW

File: bndtools.core/src/org/bndtools/core/resolve/ui/ResolutionSuccessPanel.java [hunk 1]
```diff
@@ -75,21 +76,19 @@ public ResolutionSuccessPanel(BndEditModel model, ResolutionResultPresenter pres
...
-
+    
     public void createControl(Composite parent) {
         FormToolkit toolkit = new FormToolkit(parent.getDisplay());
-        composite = toolkit.createComposite(parent);
-
-        GridLayout layout = new GridLayout(1, false);
-        composite.setLayout(layout);
-
-        SashForm form = new SashForm(composite, SWT.VERTICAL);
```
Comment (gamerson) [inline bndtools.core/src/org/bndtools/core/resolve/ui/ResolutionSuccessPanel.java]: We can't remove the sash form, this is what gives users ability to grab the border between the sections and manually expand/contract them.  So can you reintroduce the sash form into the hierarchy?

File: bndtools.core/src/org/bndtools/core/resolve/ui/ResolutionSuccessPanel.java [hunk 2]
```diff
@@ -75,21 +76,19 @@ public ResolutionSuccessPanel(BndEditModel model, ResolutionResultPresenter pres
...
-        form.setLayout(new GridLayout(1, false));
-        form.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+        ScrolledForm form = toolkit.createScrolledForm(parent);
+        composite = form;
 
+        GridLayout layout = new GridLayout(1, true);
```
Comment (gamerson) [inline bndtools.core/src/org/bndtools/core/resolve/ui/ResolutionSuccessPanel.java]: This layout object doesn't get used.

**Comment by gamerson** [comment]
Hey @wodencafe we were pretty close here, just checking to see if you were able to make the updates.

**Comment by wodencafe** [comment]
Hi @gamerson, I have been trying to get the sections to both auto size the others when collapsed, and still be draggable sash forms. SWT is not cooperating, it will either be draggable or auto-size capable, but not both.

Do you know of anywhere in eclipse that has done this correctly, that I can look at and see what I'm doing differently?

**Comment by gamerson** [comment]
@wodencafe can you send me what you have and I can take a look?  We may have to regroup here.

**Comment by gamerson** [comment]
@wodencafe ping :)

**Comment by bjhargrave** [comment]
Closing. Please open a new PR if you are able to address the comments raised.

**Pull request closed without merging**